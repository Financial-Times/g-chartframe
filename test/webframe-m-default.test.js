import * as fs from 'fs';
import * as d3 from 'd3-selection';
import jsdom from 'jsdom';
import webframeMDefault from '../src/webframe-m-default';

test('webframe-m-default renders as expected with defaults', () => {
    const { JSDOM } = jsdom;
    const defaultFrame = webframeMDefault();
    const dom = new JSDOM(fs.readFileSync('test/scaffold.html'));
    const chartContainer = d3.select(dom.window.document.querySelector('svg'));
    chartContainer.call(defaultFrame);
    expect(chartContainer.node()).toMatchSnapshot();
});
