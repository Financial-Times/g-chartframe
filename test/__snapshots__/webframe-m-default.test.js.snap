// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webframe-m-default renders as expected with defaults 1`] = `
<svg
  aria-labelledby="ft-webgraphic-m-default__chart-a11y-title ft-webgraphic-m-default__chart-a11y-desc"
  class="ft-webgraphic-m-default"
  font-family="MetricWeb,sans-serif"
  height="500"
  role="img"
  viewBox="0 0 700 500"
  width="700"
  xmlns="http://www.w3.org/2000/svg"
>
  <title
    id="ft-webgraphic-m-default__chart-a11y-title"
  >
    A chart
  </title>
  <desc
    id="ft-webgraphic-m-default__chart-a11y-desc"
  >
    A graphic by the Financial Times
  </desc>
  <rect
    class="chart-background"
    fill="#FFF1E0"
    height="500"
    id="chart-background"
    role="presentation"
    width="700"
    x="0"
    y="0"
  />
  <rect
    class="black-bar"
    height="4"
    style="fill: #000;"
    width="60"
  />
  <text
    class="chart-title"
    id="ft-webgraphic-m-defaulttitle"
  >
    <tspan
      fill="#000"
      font-family="MetricWeb,sans-serif"
      font-size="28"
      font-weight="400"
      x="1"
      y="32"
    >
      Title: A description of the charts purpose
    </tspan>
  </text>
  <text
    class="chart-subtitle"
    id="ft-webgraphic-m-defaultsubtitle"
  >
    <tspan
      fill="#66605C"
      font-family="MetricWeb,sans-serif"
      font-size="24"
      id="ft-webgraphic-m-defaultsubtitle"
      x="1"
      y="68"
    >
      some supporting information, units perhaps
    </tspan>
  </text>
  <text
    class="chart-source"
    id="ft-webgraphic-m-defaultsource"
  >
    <tspan
      fill="#66605C"
      font-family="MetricWeb,sans-serif"
      font-size="24"
      id="ft-webgraphic-m-defaultsource"
      x="1"
      y="446"
    >
      Source: research
    </tspan>
  </text>
  <text
    class="chart-copyright"
  >
    <tspan
      fill="#66605C"
      font-family="MetricWeb,sans-serif"
      font-size="14px"
      font-style="italic"
      x="1"
      y="464"
    >
      Â© FT
    </tspan>
  </text>
  <g
    class="chart-watermark"
    role="presentation"
    transform="translate(536,490) scale(1) "
  />
  <g
    class="chart-plot"
    role="presentation"
    transform="translate(20,108)"
  />
</svg>
`;
